nav2_mppi_controller/src/noise_generator.cpp

```diff
@@ -107,17 +113,10 @@ void NoiseGenerator::noiseThread()
 void NoiseGenerator::generateNoisedControls()
 {
   auto & s = settings_;
-
-  xt::noalias(noises_vx_) = xt::random::randn<float>(
-    {s.batch_size, s.time_steps}, 0.0f,
-    s.sampling_std.vx);
-  xt::noalias(noises_wz_) = xt::random::randn<float>(
-    {s.batch_size, s.time_steps}, 0.0f,
-    s.sampling_std.wz);
-  if (is_holonomic_) {
-    xt::noalias(noises_vy_) = xt::random::randn<float>(
-      {s.batch_size, s.time_steps}, 0.0f,
-      s.sampling_std.vy);
+  noises_vx_ = Eigen::ArrayXXf::NullaryExpr(s.batch_size, s.time_steps, [&] () {return ndistribution_vx_(generator_);});
```

Comment:
Does this perform a copy into the variable? Is there a way to do this without that (i.e. looping) and is that faster?

Reply:
If we assign NullaryExpr to an `auto` variable then it assigns the abstract expression. But in this case, it is being evaluated in place. But I'm not sure if it first evaluates the data in a temporary place and then copies it to the variable. 
I checked the implementation of Random() method provided by Eigen, and they're also using NullaryExpr [Refer to link](https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Core/Random.h?ref_type=heads#:~:text=%3Ctypename%20Derived%3E-,inline%20const%20typename%20DenseBase%3CDerived%3E%3A%3ARandomReturnType%20DenseBase%3CDerived%3E%3A%3ARandom(Index,%7D,-/**%20%5Creturns%20a%20random)

Reply:
I agree this look reasonable and correct :+1:  Did you try looping though and seeing if that was faster? Sometimes particular implementations are faster for particular instances vs what the core library does that works in an abstract sense (which the author might think that array could be 1M+ in size which doing something like this might be more sensible). I don't know if `NullaryExpr` is something that using OpenMP/TBB have parallelized execution for.

Reply:
Ditto here

Reply:
I've tried looping instead of NullaryExpr. Both are almost equal in timing, but NullaryExpr is a little bit faster. Using OpenMP for a 2000 * 50(or even 5000 * 100) array is slower than normal sequential for loop.